std/typecons.d

```diff
@@ -2944,6 +2959,19 @@ auto nullable(T)(T t)
     assert(ntts.to!string() == "2.5");
 }
 
+// Bugzilla 14477
+@safe unittest
+{
+    static struct DisabledDefaultConstructor
+    {
+        @disable this();
+        this(int i) { }
+    }
+    Nullable!DisabledDefaultConstructor var;
+    var = DisabledDefaultConstructor(5);
+    var.nullify;
+}
```

Comment:
I think it would be a good idea to add a couple of more cases here, e.g. the one with an overloaded assignment, s.t. someone in the future doesn't accidentally "fix" the test.

Reply:
Wouldn't that be testing internal properties? Nullable with overloaded assignment *could* work, in theory, if e.g. emplace supported const; the fact that it doesn't work now is merely because I didn't want to put in the (significant) effort to support it correctly.